include/xrpl/protocol/STObject.h

```diff
@@ -244,6 +244,9 @@ class STObject : public STBase, public CountedObject<STObject>
     getFieldPathSet(SField const& field) const;
     STVector256 const&
     getFieldV256(SField const& field) const;
+    // If not found, returns an object constructed with the given field
+    STObject
+    getFieldObject(SField const& field) const;
```

Comment:
I think the name getFieldObject is technically clear, but it hides an important nuance: it constructs a new object if the field doesn’t exist, which is unexpected behaviour for a “get” method. That violates the principle of least surprise.

I'd name it `getOrCreateFieldObject`